runtime/lua/vim/lsp.lua

```diff
@@ -1093,19 +1093,29 @@ function lsp.start_client(config)
     end
   end
 
+  ---@private
+  -- Determines whether the given option can be set by `set_defaults`.
+  local function is_empty_or_default(bufnr, option)
+    if vim.bo[bufnr][option] == '' then
+      return true
+    end
+    local last_set_from = vim.fn.gettext('\n\tLast set from ')
+    return not not vim.fn.execute('verbose set ' .. option .. '?'):match(last_set_from .. '.*ftplugin')
```

Comment:
I'm wondering if we should be more strict here, perhaps we could match against '.*/runtime/ftplugin' or even $VIMRUNTIME because ftplugin may be part of user's configuration.

Reply:
checking `$VIMRUNTIME` seems like a good idea.

Also need a test exercising this. Hopefully just add some checks in here : https://github.com/neovim/neovim/blob/2e450efb9525308708dd9e26d6ea1d029ac781b8/test/functional/plugin/lsp_spec.lua#L278 

or, alas, maybe need to copy-paste that test and create a `it('sets defaults')` test.

Reply:
I added tests and changed the code to check for vimruntime.